src/test/java/at/ac/tuwien/kr/alpha/solver/AtomCounterTests.java

```diff
@@ -0,0 +1,140 @@
+		expectGetNumberOfAtoms(atomCounter, RuleAtom.class, 0);
+
+		createBasicAtom1();
+		createBasicAtom2();
+		createAggregateAtom();
+		createChoiceAtom();
+		createRuleAtom();
+
+		expectGetNumberOfAtoms(atomCounter, BasicAtom.class, 2);
+		expectGetNumberOfAtoms(atomCounter, AggregateAtom.class, 1);
+		expectGetNumberOfAtoms(atomCounter, ChoiceAtom.class, 1);
+		expectGetNumberOfAtoms(atomCounter, RuleAtom.class, 1);
+	}
+
+	@Test
+	public void testGetStatsByType() throws NoSuchMethodException {
+		final AtomCounter atomCounter = atomStore.getAtomCounter();
+
+		createBasicAtom1();
+		createBasicAtom2();
+		createAggregateAtom();
+		createChoiceAtom();
+		createRuleAtom();
+
+		expectGetStatsByType(atomCounter, BasicAtom.class, 2);
+		expectGetStatsByType(atomCounter, AggregateAtom.class, 1);
+		expectGetStatsByType(atomCounter, ChoiceAtom.class, 1);
+		expectGetStatsByType(atomCounter, RuleAtom.class, 1);
+	}
+
+	private void createBasicAtom1() {
+		atomStore.putIfAbsent(new BasicAtom(Predicate.getInstance("p", 0)));
+	}
+
+	private void createBasicAtom2() {
+		atomStore.putIfAbsent(new BasicAtom(Predicate.getInstance("q", 1), ConstantTerm.getInstance(1)));
+	}
+
+	private void createAggregateAtom() {
+		final ConstantTerm<Integer> c1 = ConstantTerm.getInstance(1);
+		final ConstantTerm<Integer> c2 = ConstantTerm.getInstance(2);
+		final ConstantTerm<Integer> c3 = ConstantTerm.getInstance(3);
+		List<Term> basicTerms = Arrays.asList(c1, c2, c3);
+		AggregateAtom.AggregateElement aggregateElement = new AggregateAtom.AggregateElement(basicTerms, Collections.singletonList(new BasicAtom(Predicate.getInstance("p", 3), c1, c2, c3).toLiteral()));
+		atomStore.putIfAbsent(new AggregateAtom(ComparisonOperator.LE, c1, null, null, AggregateAtom.AGGREGATEFUNCTION.COUNT, Collections.singletonList(aggregateElement)));
+	}
+
+	private void createChoiceAtom() {
+		atomStore.putIfAbsent(ChoiceAtom.on(1));
+	}
+
+	private void createRuleAtom() {
+		Atom atomAA = new BasicAtom(Predicate.getInstance("aa", 0));
+		Rule ruleAA = new Rule(new DisjunctiveHead(Collections.singletonList(atomAA)), Collections.singletonList(new BasicAtom(Predicate.getInstance("bb", 0)).toLiteral(false)));
+		atomStore.putIfAbsent(new RuleAtom(NonGroundRule.constructNonGroundRule(ruleAA), new Substitution()));
+	}
+
+	private void expectGetNumberOfAtoms(AtomCounter atomCounter, Class<? extends Atom> classOfAtoms, int expectedNumber) {
+		assertEquals("Unexpected number of " + classOfAtoms.getSimpleName() + "s", expectedNumber, atomCounter.getNumberOfAtoms(classOfAtoms));
+	}
+
+	private void expectGetStatsByType(AtomCounter atomCounter, Class<? extends Atom> classOfAtoms, int expectedNumber) {
+		assertTrue("Expected number of " + classOfAtoms.getSimpleName() + "s not contained in stats string", atomCounter.getStatsByType().contains(classOfAtoms.getSimpleName() + ": " + expectedNumber));
+	}
+
+	@at.ac.tuwien.kr.alpha.common.atoms.external.Predicate
```

Comment:
Is this actually used? I think checking of the external atoms is a good idea and should be included in the tests.

Reply:
Oh, I forgot to delete this. How could external atoms be tested here? `AtomStoreImpl#putIfAbsent` does not work for them because `ExternalAtom#isGround` returns false.

Reply:
I took a quick look at that  - It seems to me that ```ExternalAtom```s never come into contact with an ```AtomStore``` anyway since they're ```FixedInterpretationLiterals``` (or rather the enclosing Literals are). That means externals are fully evaluated by the grounder (Lines 418-432 in ```NaiveGrounder.bindNextAtomInRule```) and are never shown to the solver, meaning testing them here isn't necessary.

Reply:
Thanks, I removed the unused code.